c2corg_ui/static/js/auth.js

```diff
@@ -0,0 +1,176 @@
+goog.provide('app.AuthController');
+goog.provide('app.authDirective');
+
+goog.require('app');
+goog.require('app.Authentication');
+goog.require('ngeo.Location');
+
+
+/**
+ * This directive is used to manage the login/register forms.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.authDirective = function() {
+  return {
+    restrict: 'A',
+    scope: true,
+    controller: 'appAuthController',
+    controllerAs: 'authCtrl',
+    bindToController: true
+  };
+};
+
+
+app.module.directive('appAuth', app.authDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.$http} $http
+ * @param {string} apiUrl Base URL of the API.
+ * @param {app.Authentication} appAuthentication
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.AuthController = function($scope, $http, apiUrl, appAuthentication,
+    ngeoLocation) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.apiUrl_ = apiUrl;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.appAuthentication_ = appAuthentication;
+
+  /**
+   * @type {ngeo.Location}
+   * @private
+   */
+  this.ngeoLocation_ = ngeoLocation;
+};
+
+
+/**
+ * @export
+ */
+app.AuthController.prototype.login = function() {
+  this.http_.post(this.buildUrl_('login'), this.scope_['login'], {
+    'headers': {
+      'Content-Type': 'application/json; charset=UTF-8',
+      'Accept': 'application/json'
+    }
+  }).then(
+      goog.bind(this.successLogin_, this),
+      goog.bind(this.errorLogin_, this)
+  );
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.AuthController.prototype.successLogin_ = function(response) {
+  this.appAuthentication_.setUserData(response['data']);
+  // redirect to previous page
+  var url_from = this.ngeoLocation_.hasParam('from') ?
+      goog.string.urlDecode(this.ngeoLocation_.getParam('from')) : '/';
```

Comment:
``goog.string.urlDecode`` is based upon ``decodeURIComponent`` but adds an additional ``+`` replacement:
https://closure-library.googlecode.com/git-history/docs/local_closure_goog_string_string.js.source.html#line437
I think it is safe to use the standard function instead.
Ditto for the encoding function, defined a few lines above.

Done at f84e3e1